src/inc/quic_driver_connection.h

```diff
@@ -0,0 +1,373 @@
+/*++
+
+    Copyright (c) Microsoft Corporation.
+    Licensed under the MIT License.
+
+--*/
+
+#pragma once
+
+#define QUIC_TEST_APIS 1
+#include "quic_platform.h"
+#include "quic_trace.h"
+
+#ifdef _WIN32
+
+class QuicDriverService {
+    SC_HANDLE ScmHandle;
+    SC_HANDLE ServiceHandle;
+public:
+    QuicDriverService() :
+        ScmHandle(nullptr),
+        ServiceHandle(nullptr) {
+    }
+    bool Initialize() {
+        uint32_t Error;
+        ScmHandle = OpenSCManager(nullptr, nullptr, SC_MANAGER_ALL_ACCESS);
+        if (ScmHandle == nullptr) {
+            Error = GetLastError();
+            QuicTraceEvent(
+                LibraryErrorStatus,
+                "[ lib] ERROR, %u, %s.",
+                Error,
+                "GetFullPathName failed");
+            return false;
+        }
+    QueryService:
+        ServiceHandle =
+            OpenServiceA(
+                ScmHandle,
+                QUIC_TEST_DRIVER_NAME,
```

Comment:
This needs to be passed in as an arg, right?

Reply:
No, because its a macro required to be defined. Lots of places in here depend on being able to combine that macro with another string.

Reply:
But calling it `QUIC_TEST_DRIVER_NAME` isn't great if you're using this for perf.

Reply:
Yeah, the more I think about it, I really prefer this file not to rely on any outside #defines. Please make these take the parameters as arguments. The couple of places that are combining macros, just remove the macro part and have a more generic error string.

Reply:
Ok. I just renamed it to `QUIC_DRIVER_NAME`